utils.js

```diff
@@ -140,17 +157,41 @@ let utils = {
      * @param {Object} options.credentials Request credentials.
      * @param {string} options.version API version to make the request against.
      * @param {string} options.xthorn Value of the X-Thorn header.
+     * @param {number} [options.retries=0] Number of times to retry.
      * @return {ChakramPromise} Promise resolving to the result of the request.
      *
      * @private
      */
     _oauthRequest: function _oauthRequest(options) {
+        let attempts = 1;
         let url = utils.constructUrl(options.version, 'oauth2/token');
-        return chakram.post(url, options.credentials, {
-            headers: {
-                'X-Thorn': options.xthorn,
-            },
-        });
+        options.retries = options.retries || 0;
+        console.log("retries: " + options.retries);
```

Comment:
I left this in on purpose.

We do admin logins in (IIUC) 3 classes:

* `Fixtures`
* `MetadataFetcher`
* `UserAgent`, in the case of `Agent.ADMIN`

I've only fixed the first of these. The question is: which ones do we want to fix?

IMO, we should definitely fix the first and the second. The third is still done manually; I decided not to change it at the moment because it has some highly complex login behavior.